Add files via upload

antonychiu2 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

antonychiu2 committed:
Add files via upload

Commit:
Auto commit for SQL_Injection-aa8bda90-a690-40da-b7f2-151285709532

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/aa8bda90-a690-40da-b7f2-151285709532)

Commit:
Auto commit for SQL_Injection-4b8927a6-9961-408e-8058-3c5266eff065

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/4b8927a6-9961-408e-8058-3c5266eff065)

Commit:
Auto commit for SQL_Injection-dc05aa65-f215-4469-9c9f-f952342d5608

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/dc05aa65-f215-4469-9c9f-f952342d5608)

Commit:
Auto commit for SQL_Injection-86eb54a8-0f13-425a-89a7-5588901ba1f9

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/86eb54a8-0f13-425a-89a7-5588901ba1f9)

Commit:
Auto commit for SQL_Injection-93728785-d71e-4571-a033-6a8375a0ed28

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/93728785-d71e-4571-a033-6a8375a0ed28)

Commit:
Auto commit for SQL_Injection-e902c92c-61dd-42e8-8fca-a38726992749

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/e902c92c-61dd-42e8-8fca-a38726992749)

Commit:
Auto commit for SQL_Injection-97762d83-ade7-40ab-90a1-2a0f094afdad

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/97762d83-ade7-40ab-90a1-2a0f094afdad)

Commit:
Auto commit for SQL_Injection-9f5b6bd7-e62c-4607-905e-8f42f60407a2

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/9f5b6bd7-e62c-4607-905e-8f42f60407a2)

Commit:
Auto commit for SQL_Injection-48d315f7-93b1-472f-84a7-e59e51408d37

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/48d315f7-93b1-472f-84a7-e59e51408d37)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -52,7 +52,7 @@
         let AWSregex = new RegExp('.*' + search + '$');
         if (listOnly) {
             promise = tpdModel.find({
-                'tld': { '$regex': AWSregex },
+                'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
             promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/e902c92c-61dd-42e8-8fca-a38726992749)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -31,7 +31,7 @@
     getGraphDocsByZone: function (zone) {
         let limitQuery = { 'docs': 1 };
         return graphDocsModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/93728785-d71e-4571-a033-6a8375a0ed28)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -38,6 +38,6 @@
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
-        return (graphDataModel.countDocuments({ 'zone': zone }).exec());
+        return (graphDataModel.countDocuments({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec());
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/48d315f7-93b1-472f-84a7-e59e51408d37)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -34,7 +34,7 @@
     getGraphDataByZone: function (zone) {
         let limitQuery = { 'data': 1, 'errs': 1 };
         return graphDataModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs_data.js b/web_server/config/models/graphs_data.js
--- a/web_server/config/models/graphs_data.js
+++ b/web_server/config/models/graphs_data.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/dc05aa65-f215-4469-9c9f-f952342d5608)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -31,6 +31,6 @@
 module.exports = {
     CertGraphModel: certGraphModel,
     getGraphDataByZone: function (zone) {
-        return certGraphModel.findOne({ 'zone': zone }).exec();
+        return certGraphModel.findOne({ 'zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/cert_graphs.js b/web_server/config/models/cert_graphs.js
--- a/web_server/config/models/cert_graphs.js
+++ b/web_server/config/models/cert_graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/86eb54a8-0f13-425a-89a7-5588901ba1f9)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -45,7 +45,7 @@
         return (promise);
     },
     getTPDsByTPD: function (tpd) {
-        return tpdModel.findOne({ 'tld': tpd });
+        return tpdModel.findOne({ 'tld': mongoSanitize.sanitize({ data: tpd }).data });
     },
     getTPDsByWildcard: function (search, listOnly) {
         let promise;
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/4b8927a6-9961-408e-8058-3c5266eff065)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -34,7 +34,7 @@
     getGraphLinksByZone: function (zone) {
         let limitQuery = { 'links': 1 };
         return graphLinksModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
 };
 
```

```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/aa8bda90-a690-40da-b7f2-151285709532)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -36,7 +36,7 @@
     getTPDsByZone: function (zone, listOnly) {
         let promise;
         if (!listOnly) {
-            promise = tpdModel.find({ 'zones.zone': zone }).exec();
+            promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
                 'zones.zone': zone,
 
```

```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/97762d83-ade7-40ab-90a1-2a0f094afdad)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -34,7 +34,7 @@
     getGraphConfigByZone: function (zone) {
         let limitQuery = { 'config': 1 };
         return graphModel.findOne({
-            'zone': zone,
+            'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
     },
     getGraphCountByZone: function (zone) {
 
```

```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -12,6 +12,7 @@
  * governing permissions and limitations under the License.
  */
 
+const mongoSanitize = require('express-mongo-sanitize');
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
 
```

```diff
diff --git a/web_server/package.json b/web_server/package.json
--- a/web_server/package.json
+++ b/web_server/package.json
@@ -30,6 +30,7 @@
     "ejs": "^3.1.7",
     "escape-string-regexp": "^1.0.5",
     "express": "^4.16.4",
+    "express-mongo-sanitize": ">=2.2.0",
     "express-session": "^1.17.1",
     "jquery": "^3.3.1",
     "mongoose": "^6.2.5",
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/43ef8fb5-744e-493c-b4c0-ae204a9a4231/fix/9f5b6bd7-e62c-4607-905e-8f42f60407a2)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Commit:
Auto commit for SQL_Injection-ec288f83-79f5-4f02-bfe1-71b9c8874fce

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/ec288f83-79f5-4f02-bfe1-71b9c8874fce)

Commit:
Auto commit for SQL_Injection-a4179f21-bae7-4228-ae7c-51f1db19b11f

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/a4179f21-bae7-4228-ae7c-51f1db19b11f)

Commit:
Auto commit for SQL_Injection-39077c3a-14b9-4992-832b-719f78f6c8b0

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/39077c3a-14b9-4992-832b-719f78f6c8b0)

Commit:
Auto commit for SQL_Injection-2036f00b-3412-40bc-92d3-2c2bc51e0fcf

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/2036f00b-3412-40bc-92d3-2c2bc51e0fcf)

Commit:
Auto commit for SQL_Injection-4740f498-d737-4ba0-ab81-1b8bf41d7662

This change fixes a **high severity** (🚩) **SQL Injection** issue reported by **Snyk**.

## Issue description
SQL Injection allows attackers to execute malicious SQL queries by manipulating input data. This can result in unauthorized access to sensitive data, data manipulation, or even complete database compromise.

## Fix instructions
Use parameterized queries or prepared statements to sanitize user input and prevent manipulation of the SQL query.

[More info and fix customization are available in the Mobb platform](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/4740f498-d737-4ba0-ab81-1b8bf41d7662)

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs.js b/web_server/config/models/graphs.js
--- a/web_server/config/models/graphs.js
+++ b/web_server/config/models/graphs.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphModel: graphModel,
     getGraphConfigByZone: function (zone) {
-        let limitQuery = { 'config': 1 };
+        let limitQuery = { 'config': mongoSanitize.sanitize({ data: 1 }).data };
         return graphModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/4740f498-d737-4ba0-ab81-1b8bf41d7662)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_links.js b/web_server/config/models/graphs_links.js
--- a/web_server/config/models/graphs_links.js
+++ b/web_server/config/models/graphs_links.js
@@ -33,7 +33,7 @@
 module.exports = {
     GraphLinksModel: graphLinksModel,
     getGraphLinksByZone: function (zone) {
-        let limitQuery = { 'links': 1 };
+        let limitQuery = { 'links': mongoSanitize.sanitize({ data: 1 }).data };
         return graphLinksModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/2036f00b-3412-40bc-92d3-2c2bc51e0fcf)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/graphs_docs.js b/web_server/config/models/graphs_docs.js
--- a/web_server/config/models/graphs_docs.js
+++ b/web_server/config/models/graphs_docs.js
@@ -30,7 +30,7 @@
 module.exports = {
     GraphDocsModel: graphDocsModel,
     getGraphDocsByZone: function (zone) {
-        let limitQuery = { 'docs': 1 };
+        let limitQuery = { 'docs': mongoSanitize.sanitize({ data: 1 }).data };
         return graphDocsModel.findOne({
             'zone': mongoSanitize.sanitize({ data: zone }).data,
         }, limitQuery).exec();
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/a4179f21-bae7-4228-ae7c-51f1db19b11f)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -40,7 +40,7 @@
             promise = tpdModel.find({ 'zones.zone': mongoSanitize.sanitize({ data: zone }).data }).exec();
         } else {
             promise = tpdModel.find({
-                'zones.zone': zone,
+                'zones.zone': mongoSanitize.sanitize({ data: zone }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         }
         return (promise);
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/ec288f83-79f5-4f02-bfe1-71b9c8874fce)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.

Mobbantony commented:
# ![image](https://app.mobb.ai/gh-action/Logo_Rounded_Icon.svg) SQL Injection fix has been committed ✅
```diff
diff --git a/web_server/config/models/tpds.js b/web_server/config/models/tpds.js
--- a/web_server/config/models/tpds.js
+++ b/web_server/config/models/tpds.js
@@ -56,7 +56,7 @@
                 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data,
             }, { 'tld': 1, 'zones.zone': 1 }).exec();
         } else {
-            promise = tpdModel.find({ 'tld': { '$regex': AWSregex } }).exec();
+            promise = tpdModel.find({ 'tld': mongoSanitize.sanitize({ data: { '$regex': AWSregex } }).data }).exec();
         }
         return (promise);
     },
 
```

[Learn more and fine tune the fix](https://app.mobb.ai/organization/afc837fb-ecb7-4b3f-9eda-127127cca2c2/project/927e9c7f-4a09-45b7-936b-735ef095f3c4/report/7710b539-bd4b-4bd1-8d9f-a27d24ed95f4/fix/39077c3a-14b9-4992-832b-719f78f6c8b0)

This fix was committed automatically according to a defined Fix Policy. Visit the Mobb app settings to control this policy.
